server/src/test-integration/java/com/thoughtworks/go/server/persistence/MaterialRepositoryIntegrationTest.java

```diff
@@ -1316,6 +1320,162 @@ public Object doInTransaction(TransactionStatus status) {
         assertThat(repo.getTotalModificationsFor(materialInstance), is(new Long(count + 1)));
     }
 
+    @Test
+    public void shouldFetchModificationsWithMaterial() {
+        MaterialRevisions materialRevisions = new MaterialRevisions();
+        SvnMaterial material = MaterialsMother.svnMaterial("http://username:password@localhost");
+        List<Modification> modificationList = getModifications(1);
```

Comment:
Since we testing the behavior of fetching the latest modification, our setup should ensure that we have more than 1 modification for a material.